crates/ruma-events/src/rtc/notification.rs

```diff
@@ -0,0 +1,213 @@
+    pub mentions: Option<Mentions>,
+
+    /// Optional reference to the related `m.rtc.member` event.
+    #[serde(rename = "m.relates_to", skip_serializing_if = "Option::is_none")]
+    pub relates_to: Option<Reference>,
+
+    /// How this notification should notify the receiver.
+    pub notification_type: NotificationType,
+}
+
+impl RtcNotificationEventContent {
+    /// Creates a new `RtcNotificationEventContent` with the given configuration.
+    pub fn new(
+        sender_ts: MilliSecondsSinceUnixEpoch,
+        lifetime: Duration,
+        notification_type: NotificationType,
+    ) -> Self {
+        Self { sender_ts, lifetime, mentions: None, relates_to: None, notification_type }
+    }
+
+    /// Calculates the timestamp at which this notification is considered invalid.
+    /// This calculation is based on MSC4075 and tries to use the `sender_ts` as the starting point
+    /// and the `lifetime` as the duration for which the notification is valid.
+    ///
+    /// The `sender_ts` cannot be trusted since it is a generated value by the sending client.
+    /// To mitigate issue because of misconfigured client clocks, the MSC requires
+    /// that the `origin_server_ts` is used as the starting point if the difference is large.
+    ///
+    /// # Arguments:
+    ///
+    /// - `max_sender_ts_offset` is the maximum allowed offset between the two timestamps. (default
+    ///   20s)
+    /// - `origin_server_ts` has to be set to the origin_server_ts from the event containing this
+    ///   event content.
+    ///
+    /// # Examples
+    /// To start a timer until this client should stop ringing for this notification:
+    /// `let duration_ring =
+    /// MilliSecondsSinceUnixEpoch::now().saturated_sub(content.invalid_ts(event.origin_server_ts(),
+    /// None));`
+    pub fn invalid_ts(
+        &self,
+        origin_server_ts: MilliSecondsSinceUnixEpoch,
+        max_sender_ts_offset: Option<u32>,
+    ) -> MilliSecondsSinceUnixEpoch {
+        let (larger, smaller) = if self.sender_ts.get() > origin_server_ts.get() {
+            (self.sender_ts.get(), origin_server_ts.get())
+        } else {
+            (origin_server_ts.get(), self.sender_ts.get())
+        };
+        let use_origin_server_ts =
+            larger.saturating_sub(smaller) > max_sender_ts_offset.unwrap_or(20_000).into();
+        let start_ts =
+            if use_origin_server_ts { origin_server_ts.get() } else { self.sender_ts.get() };
+        MilliSecondsSinceUnixEpoch(
+            start_ts.saturating_add(UInt::from(self.lifetime.as_millis() as u32)),
+        )
+    }
+}
+
+/// How this notification should notify the receiver.
+#[derive(Clone, StringEnum)]
+#[cfg_attr(not(ruma_unstable_exhaustive_types), non_exhaustive)]
+#[ruma_enum(rename_all = "snake_case")]
+pub enum NotificationType {
+    /// The receiving client should ring with an audible sound.
+    Ring,
+    /// The receiving client should display a visual notification.
```

Comment:
Leave an empty line between variants:

```suggestion

    /// The receiving client should display a visual notification.
```